### ISSUE TITLE
Executor trait `spawn_bg` method violates runtime-agnostic abstraction with fragile drop-based detachment

### ISSUE THREAD

**Issue by GrantSparks** [GrantSparks/grafton-visca#409]
## Background
The grafton-visca library maintains a runtime-agnostic design through its `Executor` trait, which abstracts over different async runtimes (tokio, async-std, smol). The library spawns critical background tasks like the runtime loop for managing VISCA communication, which must continue running independently after being spawned. The `spawn_bg` method is provided as a convenience method specifically for these fire-and-forget background tasks.

## Problem
The default implementation of `spawn_bg` in the `Executor` trait uses a naive approach of dropping the join handle returned by `spawn()`. This creates a fundamental architectural flaw:

```rust
fn spawn_bg<F>(&self, fut: F) 
where F: Future + Send + 'static, F::Output: Send + 'static,
{
    drop(self.spawn(fut));  // Assumes dropping = detachment
}
```

This assumption is incorrect for several executor implementations:
- **DeterministicExecutor**: Dropping the join handle cancels the task entirely
- **async-executor**: Tasks require explicit `.detach()` to survive dropped handles
- **Custom executors**: May have different ownership semantics

The issue manifests in production code at `src/runtime/handle.rs:502` where `spawn_runtime_loop` uses `spawn_bg` to start the critical runtime communication loop. If the executor's join handle owns the task, the background loop is immediately cancelled, causing complete communication failure.

## Evidence

1. **DeterministicExecutor override requirement** (`src/testing/testkit/deterministic_executor.rs:811-817`):
```rust
fn spawn_bg<F>(&self, fut: F) {
    self.executor.spawn(fut).detach();  // Must explicitly detach
}
```

2. **Test utilities workaround** (`src/testing/testkit/deterministic_executor.rs:854-859`):
```rust
impl ExecutorExt for DeterministicExecutor {
    fn spawn_detached<F>(&self, fut: F) {
        self.executor.spawn(fut).detach();  // Separate trait needed
    }
}
```

3. **Runtime loop spawning** (`src/runtime/handle.rs:502`):
```rust
executor.spawn_bg(async move {
    let _ = runtime_loop_with_config::<P, T, E>(...).await;
});
```

## Proposed Change

Replace the drop-based approach with an explicit detachment method in the Executor trait:

```rust
pub trait Executor: Clone + Send + Sync + 'static {
    // ... existing associated types ...
    
    /// Type indicating whether tasks require explicit detachment.
    /// Set to () if dropping join handle detaches, or a detachment token otherwise.
    type Detach: Send + 'static;
    
    /// Spawn a task and return both join handle and optional detachment token.
    fn spawn_with_detach<F>(&self, fut: F) -> (Self::Join<F::Output>, Self::Detach)
    where F: Future + Send + 'static, F::Output: Send + 'static;
    
    /// Default spawn delegates to spawn_with_detach, discarding detach token.
    fn spawn<F>(&self, fut: F) -> Self::Join<F::Output>
    where F: Future + Send + 'static, F::Output: Send + 'static,
    {
        self.spawn_with_detach(fut).0
    }
    
    /// Spawn background task with guaranteed detachment.
    fn spawn_bg<F>(&self, fut: F)
    where F: Future + Send + 'static, F::Output: Send + 'static,
    {
        let (handle, detach_token) = self.spawn_with_detach(fut);
        drop(handle);
        drop(detach_token);  // Dropping detach token ensures task continues
    }
}
```

Implementation examples:

```rust
// Tokio - no explicit detachment needed
impl Executor for TokioExecutor {
    type Detach = ();
    
    fn spawn_with_detach<F>(&self, fut: F) -> (Self::Join<F::Output>, ()) {
        (TokioJoin(self.handle.spawn(fut)), ())
    }
}

// DeterministicExecutor - returns detachment handle
impl Executor for DeterministicExecutor {
    type Detach = DetachmentHandle;
    
    fn spawn_with_detach<F>(&self, fut: F) -> (DetJoin, DetachmentHandle) {
        let task = self.executor.spawn(fut);
        let detach_handle = DetachmentHandle(task.clone());
        (DetJoin(task), detach_handle)
    }
}

impl Drop for DetachmentHandle {
    fn drop(&mut self) {
        self.0.detach();  // Explicit detachment on drop
    }
}
```

## Alternatives

1. **Runtime-specific spawn_bg overrides** (Current workaround)
   - Requires each executor to override spawn_bg correctly
   - Error-prone and violates DRY principle
   - Quality argument against: Increases maintenance burden and risk of silent failures

2. **Remove spawn_bg, use spawn everywhere**
   - Forces all callers to manage join handles
   - Quality argument against: Reduces ergonomics and increases boilerplate for common pattern

3. **Add runtime detection and conditional logic**
   - Check runtime type and apply appropriate detachment
   - Quality argument against: Violates zero-cost abstraction, adds runtime overhead

The proposed solution maintains zero-cost abstraction, provides compile-time correctness, and aligns with Rust's ownership principles.

## Testing Plan

1. **Unit tests for each executor implementation**:
   - Verify spawn_bg tasks continue after handle drop
   - Test task cancellation behavior with regular spawn
   
2. **Integration tests**:
   - Verify runtime loop continues running with spawn_bg
   - Test socket manager background task persistence
   
3. **Deterministic executor tests**:
   - Confirm background tasks run in block_on_bg
   - Verify no task leakage or premature cancellation
   
4. **Migration validation**:
   - Update all executor implementations
   - Remove ExecutorExt trait workarounds
   - Verify no regressions in existing tests

## Backward Compatibility

This change requires updating all Executor implementations but maintains full backward compatibility through default trait methods. The migration path is straightforward and type-safe, with compiler errors guiding any necessary updates.
